src/main/java/wooteco/subway/web/member/interceptor/BearerAuthMemberInterceptor.java

```diff
@@ -0,0 +1,95 @@
+package wooteco.subway.web.member.interceptor;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.HandlerMapping;
+import org.springframework.web.servlet.ModelAndView;
+import wooteco.subway.infra.JwtTokenProvider;
+import wooteco.subway.web.member.exception.InvalidAuthenticationException;
+import wooteco.subway.web.member.exception.NotMatchedEmailIExistInJwtException;
+import wooteco.subway.web.member.util.AuthorizationExtractor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Map;
+
+import static org.springframework.http.HttpMethod.*;
+
+@Component
+public class BearerAuthMemberInterceptor implements HandlerInterceptor {
+    private AuthorizationExtractor authExtractor;
+    private JwtTokenProvider jwtTokenProvider;
+
+    public BearerAuthMemberInterceptor(AuthorizationExtractor authExtractor, JwtTokenProvider jwtTokenProvider) {
+        this.authExtractor = authExtractor;
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request,
+                             HttpServletResponse response, Object handler) {
+        if (isPost(request)) {
+            return true;
+        }
+
+        String bearer = authExtractor.extract(request, "Bearer");
+        validateToken(bearer);
+        String email = jwtTokenProvider.getSubject(bearer);
+        request.setAttribute("requestMemberEmail", email);
+
+        if (isGet(request)) {
```

Comment:
```suggestion
        if (GET.matches(request.getMethod())) {
```
조건문을 메서드로 빼는게 꼭 좋은 방법은 아닙니다. 이해하기 힘든 조건을 메서드로 뺐을 때는 좋으나 지금처럼 단순한 경우엔 실제 조건까지 보려면 메서드를 찾아가야하는 수고가 동반되고 클래스 내 메서드들도 비대해지게 되죠. 메서드를 추출해서 발생하는 트레이드 오프를 고려해볼 여지가 있습니다.

```java
for(int i = 0; isUnderTen(i); i++ ) {
```
isUnderTen의 구현이 어떨지는 간단히 유추가 될 겁니다.